scripts/tests/cross-test-ci-all.sh

```diff
@@ -0,0 +1,93 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+# all versions to use for testing
+versions=("v0.2.1")
+>&2 echo "Running backwards-compatibility tests for versions: ${versions[*]}"
+
+# signal to downstream test scripts
+export FM_BACKWARDS_COMPATIBILITY_TEST=1
+
+function nix_build_binary_for_version() {
+  binary="$1"
+  version="$2"
+  echo "$(nix build 'github:fedimint/fedimint/'"$version"'#'"$binary" --no-link --print-out-paths)/bin/$binary"
+}
+
+function use_fed_binaries_for_version() {
+  version=$1
+  if [[ "$version" == "current" ]]; then
+    unset FM_FEDIMINTD_BASE_EXECUTABLE
+  else
+    >&2 echo "Compiling fed binaries for version $version..."
+    FM_FEDIMINTD_BASE_EXECUTABLE="$(nix_build_binary_for_version 'fedimintd' "$version")"
+    export FM_FEDIMINTD_BASE_EXECUTABLE
+  fi
+}
+
+function use_client_binaries_for_version() {
```

Comment:
I'm allowing `gateway-cli` and `gatewayd` to be on different versions since that's the only way automated tests would have caught https://github.com/fedimint/fedimint/issues/3977.

If we're assuming a gateway operator is updating the daemon at the same time as the client, do we need to ensure changes are backwards-compatible?